### PR TITLE
Remove build dir on cleanup

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

**Comment by SeaDve** [comment]
This is not the behavior though of `gnome-builder`'s clean command

**Comment by bilelmoussaoui** [comment]
> This is not the behavior though of `gnome-builder`'s clean command

What do the cleanup do in general? or in other words, how can you force a fresh rebuild?

**Comment by SeaDve** [comment]
It should be its own method then. With that, it will also delete `_build` even when modifying the manifest or changing the active manifest

**Comment by bilelmoussaoui** [comment]
> It should be its own method then. With that, it will also delete `_build` even when modifying the manifest or changing the active manifest

Right, when a manifest has changed it shouldn't call "clean" but a different method

**Comment by SeaDve** [comment]
> > It should be its own method then. With that, it will also delete `_build` even when modifying the manifest or changing the active manifest
> 
> Right, when a manifest has changed it shouldn't call "clean" but a different method

It calls only `deleteRepoDir` not `clean`, deleting the `_build` would have to be put directly on the `buildPipeline.clean`

**Review by SeaDve — APPROVED** [review]

**Comment by SeaDve** [comment]
Not big deal, but maybe put it in a method in manifest like `deleteBuildSystemBuildDir`

File: src/buildPipeline.ts [hunk 1]
```diff
@@ -26,7 +28,7 @@ export class BuildPipeline implements vscode.Disposable {
         this.manifestManager.onDidRequestRebuild(async (manifest) => {
             if (manifest === this.manifestManager.getActiveManifest()) {
                 console.log(`Manifest at ${manifest.uri.fsPath} requested a rebuild`)
-                await this.clean()
+                await manifest.deleteRepoDir()
```
Comment (SeaDve) [inline src/buildPipeline.ts:31]: This probably also have to `resetState`?

**Pull request merged**